x-pack/plugins/endpoint/common/types.ts

```diff
@@ -611,47 +611,27 @@ export enum HostPolicyResponseActionStatus {
 }
 
 /**
- * The details of a given action
+ * Host Policy Response Applied Action
  */
-export interface HostPolicyResponseActionDetails {
+export interface HostPolicyResponseAppliedAction {
+  name: string;
   status: HostPolicyResponseActionStatus;
   message: string;
 }
 
-/**
- * A known list of possible Endpoint actions
- */
-export interface HostPolicyResponseActions {
-  download_model: HostPolicyResponseActionDetails;
-  ingest_events_config: HostPolicyResponseActionDetails;
-  workflow: HostPolicyResponseActionDetails;
-  configure_elasticsearch_connection: HostPolicyResponseActionDetails;
-  configure_kernel: HostPolicyResponseActionDetails;
-  configure_logging: HostPolicyResponseActionDetails;
-  configure_malware: HostPolicyResponseActionDetails;
-  connect_kernel: HostPolicyResponseActionDetails;
-  detect_file_open_events: HostPolicyResponseActionDetails;
-  detect_file_write_events: HostPolicyResponseActionDetails;
-  detect_image_load_events: HostPolicyResponseActionDetails;
-  detect_process_events: HostPolicyResponseActionDetails;
-  download_global_artifacts: HostPolicyResponseActionDetails;
-  load_config: HostPolicyResponseActionDetails;
-  load_malware_model: HostPolicyResponseActionDetails;
-  read_elasticsearch_config: HostPolicyResponseActionDetails;
-  read_events_config: HostPolicyResponseActionDetails;
-  read_kernel_config: HostPolicyResponseActionDetails;
-  read_logging_config: HostPolicyResponseActionDetails;
-  read_malware_config: HostPolicyResponseActionDetails;
-}
-
-/**
- * policy configurations returned by the endpoint in response to a user applying a policy
- */
 export type HostPolicyResponseConfiguration = HostPolicyResponse['endpoint']['policy']['applied']['response']['configurations'];
 
 interface HostPolicyResponseConfigurationStatus {
   status: HostPolicyResponseActionStatus;
-  concerned_actions: Array<keyof HostPolicyResponseActions>;
+  concerned_actions: string[];
```

Comment:
If you implement the suggestion above, this would then be:

```suggestion
  concerned_actions: HostPolicyActionName[]
```